#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/factory/containers/protos/WoodenBox.proto"

WorldInfo {
  info [
    ""
  ]
  ERP 0.6
}
Viewpoint {
  orientation -0.5736830525110008 0.009124138957832952 0.8190265596120423 3.1155466490641297
  position 6.284666938674494 -0.29955835805276587 16.251691544741472
}
TexturedBackground {
}
TexturedBackgroundLight {
}
WoodenBox {
  translation 4.36982 1.12136 0.29000000000000004
  rotation 0 0 1 3.14159
  size 0.5 0.5 0.5
  mass 20
}
Solid {
  translation -1.12633 3.90126 0.38
  children [
    DEF BOX Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.5 0.5 0.5
      }
    }
  ]
  boundingObject USE BOX
  physics Physics {
    mass 5
    centerOfMass [
      0 0 0
    ]
  }
}
Solid {
  translation -4.62869 1.43503 0.37
  rotation 0 0 -1 0.261797
  children [
    DEF BOX Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.5 0.5 0.5
      }
    }
  ]
  name "solid(1)"
  boundingObject USE BOX
  physics Physics {
    mass 5
    centerOfMass [
      0 0 0
    ]
  }
}
Robot {
  translation 0 0 0.39
  rotation 0 1 0 0
  children [
    DEF robot_body Group {
      children [
        Transform {
          translation -0.11 0 0.18
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.4 0.3 1
                metalness 0
              }
              geometry Box {
                size 0.2 0.2 0.15
              }
            }
          ]
        }
        Transform {
          translation 0.27 -0.01 -0.0399989
          rotation 0 1 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.4 0.3 1
                metalness 0
              }
              geometry Cylinder {
                height 0.15
                radius 0.1
              }
            }
          ]
        }
        Transform {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.4 0.3 1
                metalness 0
              }
              geometry Box {
                size 0.5 0.4 0.3
              }
            }
          ]
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.11 0.25 -0.11
      }
      device [
        RotationalMotor {
          name "wheel_l1"
        }
      ]
      endPoint Solid {
        translation 0.11 0.25 -0.11
        rotation 1 0 0 1.5708
        children [
          DEF wheel_body Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
            }
            geometry Cylinder {
              height 0.1
              radius 0.1
            }
          }
        ]
        boundingObject USE wheel_body
        physics DEF wheel_physics Physics {
          mass 20
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0.11 -0.25 -0.11
      }
      device [
        RotationalMotor {
          name "wheel_r1"
        }
      ]
      endPoint Solid {
        translation 0.11 -0.25 -0.11
        rotation 1 0 0 1.5708
        children [
          USE wheel_body
        ]
        name "solid(1)"
        boundingObject USE wheel_body
        physics USE wheel_physics
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.11 -0.25 -0.11
      }
      device [
        RotationalMotor {
          name "wheel_r2"
        }
      ]
      endPoint Solid {
        translation -0.11 -0.25 -0.11
        rotation 1 0 0 1.5708
        children [
          USE wheel_body
        ]
        name "solid(2)"
        boundingObject USE wheel_body
        physics USE wheel_physics
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor -0.11 0.25 -0.11
      }
      device [
        RotationalMotor {
          name "wheel_l2"
        }
      ]
      endPoint Solid {
        translation -0.11 0.25 -0.11
        rotation 1 0 0 1.5708
        children [
          USE wheel_body
        ]
        name "solid(3)"
        boundingObject USE wheel_body
        physics USE wheel_physics
      }
    }
    DistanceSensor {
      translation 0.3 0 -0.14
      rotation 0 1 0 1.5708
      children [
        DEF ds_edge Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.05 0.05
          }
        }
      ]
      name "ds_edge"
      boundingObject USE ds_edge
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.15 1000 0
      ]
    }
    DistanceSensor {
      translation 0.350132 0 -0.0854971
      rotation 0 1 0 0
      children [
        DEF ds_obstacle Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.05 0.05
          }
        }
      ]
      name "ds_obstacle"
      boundingObject USE ds_obstacle
      physics Physics {
      }
      lookupTable [
        0 0 0
        20 1000 0
      ]
    }
  ]
  boundingObject USE robot_body
  physics Physics {
    mass 2
  }
  controller "obstacle_pusher"
}
Solid {
  translation 0 0 0.1
  children [
    DEF floor Shape {
      appearance PBRAppearance {
        baseColor 0.34 0.5 0.56
        metalness 0
      }
      geometry Box {
        size 12 12 0.1
      }
    }
  ]
  name "floor"
  boundingObject USE floor
}
